sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -66,14 +68,24 @@ public LogsQueryClient(Uri endpoint, TokenCredential credential) : this(endpoint
         /// <param name="endpoint">The service endpoint to use.</param>
         /// <param name="credential">The <see cref="TokenCredential"/> instance to use for authentication.</param>
         /// <param name="options">The <see cref="LogsQueryClientOptions"/> instance to use as client configuration.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public LogsQueryClient(Uri endpoint, TokenCredential credential, LogsQueryClientOptions options)
         {
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(endpoint, nameof(endpoint));
 
             Endpoint = endpoint;
             options ??= new LogsQueryClientOptions();
-            var authorizationScope = $"{(string.IsNullOrEmpty(options.Audience?.ToString()) ? LogsQueryAudience.AzurePublicCloud : options.Audience)}";
+            if (string.IsNullOrEmpty(options.Audience?.ToString()))
+            {
+                options.Audience = endpoint.OriginalString;
```

Comment:
@srnagar @tg-msft Does this work in all Azure Stack scenarios? I think so because the endpoint matches the Audience but just making sure :)

Reply:
Why OriginalString vs AbsoluteUri?

Reply:
OriginalString removes the last / that is automatically added in the Uri creation. Having the extra / makes the end of the scope ///.default and we only want //.default

Reply:
My understanding is that OriginalString will just have the exact string that the user passed, which could have contained things like leading spaces, trailing spaces, and url encoding characters, which I don't think we want. I would suggest adding a couple tests that construct a URI with these characters and see what the resulting behavior is.